### PR TITLE
Fix: Make the out flag print to one file using seperator

### PR DESCRIPTION
**Pull request by AhmedGrati**
# Description
This pull request solves #1539.

# Implementation
Basically, it has the exact implementation of the `helm template` command. It will output all the k8s resources in one YAML file using the `---` separator. If the user specifies a JSON output, it will return an error since we cannot have multiple JSON resources in one file.

### CODE REVIEW

File: pkg/transformer/kubernetes/k8sutils.go [hunk 1]
```diff
@@ -148,16 +148,16 @@ func getDirName(opt kobject.ConvertOptions) string {
...
 
-	bytes, _ := json.Marshal(object)
-	r.Raw = bytes
+// 	bytes, _ := json.Marshal(object)
+// 	r.Raw = bytes
 
-	return r
-}
+// 	return r
+// }
```
Comment (cdrage) [inline pkg/transformer/kubernetes/k8sutils.go:160]: remove commented out section

File: pkg/transformer/kubernetes/k8sutils.go [hunk 2]
```diff
@@ -192,36 +192,30 @@ func PrintList(objects []runtime.Object, opt kobject.ConvertOptions) error {
...
-		}
-		printVal, err := transformer.Print("", dirName, "", data, opt.ToStdout, opt.GenerateJSON, f, opt.Provider)
-		if err != nil {
-			return errors.Wrap(err, "transformer.Print failed")
+			data, err := marshal(versionedObject, opt.GenerateJSON, opt.YAMLIndent)
+			if err != nil {
+				return fmt.Errorf("error in marshalling the List: %v", err)
+			}
+			data = []byte(fmt.Sprintf("---\n%s", data))
+			printVal, err := transformer.Print("", dirName, "", data, opt.ToStdout, opt.GenerateJSON, f, opt.Provider)
```
Comment (cdrage) [inline pkg/transformer/kubernetes/k8sutils.go:213]: good, but you need to add comments "this part add `---` which unifies the file", etc.

File: pkg/transformer/kubernetes/k8sutils.go [hunk 3]
```diff
@@ -192,36 +192,30 @@ func PrintList(objects []runtime.Object, opt kobject.ConvertOptions) error {
...
+			if err != nil {
+				return errors.Wrap(err, "transformer.Print failed")
```
Comment (cdrage) [inline pkg/transformer/kubernetes/k8sutils.go:215]: I know this is outside the scope of the PR, but perhaps change this error output while you're at it :) "transformer to print to one single file failed" or something like that.

**Review by cdrage — APPROVED** [review]
boom-mind-blown

SO HAPPY THIS MAKES TESTS PASS.

Thank you! Small minor changes, but otherwise code LGTM.

Update the PR with the changes and then feel free to merge.

**Review by hangyan — APPROVED** [review]

**Comment by hangyan** [comment]
LGTM

**Pull request merged**